### ISSUE TITLE
EIT only prints first 252 Characters in event_ext text

### ISSUE THREAD

**Issue by SourceDoctor** [braice/MuMuDVB#100]
EIT.json only shows first 252 Characters in event_ext.
It should write complete TV-Programm Description instead of cutting down text.

**Comment by braice** [comment]
Hello

Thank you for your report.
Can you give the part of the json so I can see precisely which field is
concerned

Thank you

**Comment by SourceDoctor** [comment]
{
        "tag" :78,
        "len" : 1,
        "descr" : "Extended event descriptor",
        "ext_evt":{
        "descr_number" : 1 ,
        "descr_total_number" : 2,
        "language" : "deu",
        "items":[],
        "text": "Wir sind nicht alleine im Weltall. Neben Planeten und Sternen tummeln sich im luftleeren Raum jede Menge Asteroiden und Kometen. Was tun, falls die Erde irgendwann einmal durch solch ein auÃ<9f>erirdisches Geschoss ernsthaft bedroht wird? Welche MÃ¶glic"
}

As you see in "text" the Description is cutted in the middle of the sentence and in middle of a word.
The Extended Event Descriptor text should be printed complete all the time, independend of the length of the incoming text for this field

**Comment by braice** [comment]
Hello

Look at page 53 of
http://www.mumudvb.net/sites/default/files/en_300468v011301p.pdf

The text length is encoded in a 8 bit field so maximum 255 characters
without multi byte characters.

The rest of the text is in the following descriptor, see
"descr_total_number" : 2,

descriptor_number:
This 4-bit field gives the number of the descriptor. It is used to
associate information which
cannot be fitted into a single descriptor.

So you'll have to merge two of those, the EIT parser is intended to make
the tables available to the client but it is to the client app to do the
necessary parsing depending on the needs

Best

Brice

2015-09-11 22:52 GMT+02:00 SourceDoctor notifications@github.com:

**Comment by SourceDoctor** [comment]
ok thx, 
i checked this now, but in many cases if
"descr_total_number": 2     is set, 
i miss Section with "descr_number" : 2 ,

Same here:
"descr_total_number": 3     is set, 
but i miss Sections with "descr_number" : 2  and  "descr_number" : 3,

also i found an Event_descriptor with following setting:
"descr_total_number": 6
"descr_number": 8

Did i miss something here, or is it a bug?

with regards

**Comment by braice** [comment]
Hi

The tables can be long to fully get so it can be a matter of time.

For the second question, I don't know maybe you hit a bug or the provider
messed up the tables, one way to know would be to get the very very very
(-vvv) verbose logs where MuMuDVB displays all EIT packets and how he
decodes.

Best

Brice

2015-09-12 19:38 GMT+02:00 SourceDoctor notifications@github.com:

**Comment by SourceDoctor** [comment]
Hi,

i analyzed it now more precise and found out another big Problem:
(i think, this is a Bug)

"Parental rating descriptor" often lists wrong languages,
this "destroys" in many cases the JSON itself.

The languages combined looks like content of what should be in an
extended Event Descriptor.
i attached you the complete JSON and the matching Configfile for MuMuDVB.

"...
{
        "tag" :85,
        "len" : 112,
        "descr" : "Parental rating descriptor",
        "parental_rating":[
        {
            "language" : "ive",
        "rating" : "defined by broadcaster"
        },
        {
            "language" : "sum",
        "rating" : "defined by broadcaster"
        },
        {
            "language" : " au",
        "rating" : "defined by broadcaster"
        },
        {
            "language" : "kl䢬
        "rating" : "defined by broadcaster"
        },
        {
            "language" : "en,",
        "rating" : "defined by broadcaster"
        },
        {
            "language" : "die",
        "rating" : "defined by broadcaster"
        },
        {
            "language" : "den",
        "rating" : "defined by broadcaster"
        },
        {
            "language" : "Wis",
        "rating" : "defined by broadcaster"
        },
..."

Am 22.09.2015 um 21:49 schrieb Brice Dubost:

**Comment by braice** [comment]
Indeed, it's strange,

if you look page 32 of
http://www.mumudvb.net/sites/default/files/en_300468v011301p.pdf

The descriptor with tag 0x55 (85 in dec) should be a parental rating
descriptor, so either the boradcaster do something funny or I missread the
descriptor tag.

Can you give me the output in full -vvv so I will have a clearer view.

Thank you

Best

Brice

2015-09-23 22:12 GMT+02:00 SourceDoctor notifications@github.com:

**Comment by SourceDoctor** [comment]
in Attachment is a STDOUT-Log
i hope it will help to find the failure.

with regards
    Thomas

Am 24.09.2015 um 11:11 schrieb Brice Dubost:

**Comment by braice** [comment]
Github removed the attachement, can you send it to mumudvb@braice.net ?

Thank you

2015-09-24 22:35 GMT+02:00 SourceDoctor notifications@github.com:

**Comment by braice** [comment]
Hi

Thank you for your files, taking a step back it sounds to me that is an
overflow in the length of the extended event descriptor

{
        "tag" :78,
        "len" : 1,
        "descr" : "Extended event descriptor",
        "ext_evt":{
        "descr_number" : 1 ,
        "descr_total_number" : 10,
        "language" : "deu",
        "items":[],
        "text": "3D-Animationsserie, USA\/Deutschland\/Frankreich
2008GlitschikusTeam Quantum sitzt fest. In einem Haufen Weltraumschrott,
den sie genutzt haben, um einen Meteoriten von seiner Bahn abzulenken, der
auf der Erde unglaubliche Verwüstungen angerichtet "
}
},

As you see the len is specified to be one, in fact it's 2 + the length of
the descriptor, in this case 254 . . .
After that all the descriptors are wrong with a lot of unknown tags and . .
. the parental rating

I commited a fix, can you give it a try ?

Thank you again

Best regards

Brice

2015-09-25 19:19 GMT+02:00 Brice Dubost braice@braice.net:

**Comment by SourceDoctor** [comment]
Hi there,

i tried it now, and it looks much better.
i found no more "strange" Language Description,
also now all Text parts of extended Event Descriptor seems to be there now.

Thanks for fixing this Bug.

Thomas

Am 26.09.2015 um 21:36 schrieb Brice Dubost:

**Comment by braice** [comment]
Perfect !

I am glad to hear it is fixed, thank you for your help in investigating
this bug!

Best

Brice

2015-09-27 15:58 GMT+02:00 SourceDoctor notifications@github.com:

**Comment by SourceDoctor** [comment]
Hi Brice,

i think i've maybe found another Bug in EPG Handling:

On some Transponders like this one:
Astra 19.2E
freq=12187
pol=h
srate=27500
EPG seems to be buggy. It has maximal 2 extended EPG Parts and end's in
middle of a sentence.

Other Transponders like:
Astra 19.2E
freq=12544
pol=h
srate=22000
With same MuMuDVB Configuration is running well and showing extended EPG
complete, even if there are 4 or more extended EPG Parts.

i attached you a matching Part of the JSON of the Problem.
I hope you'll find the source of the Problem.

By the way. Do you remove Newlines in extended Field Information from
original EPG-Stream?
If i compare EPG Information from MuMuDVB with other Systems (including
TV-Receiver) EPG Information,
i miss all Newlines in extended EPG Information which give's the Text a
structure.
I know you correctly escape Newline in Sourcecode to transport it with
JSON, but there is no Newline in extended Field Information.
Maybe you'll can find this also.

Thanks and Greetings
    Thomas

{
        "event_id" :57583,
        "start_time day " : "2016-02-05 (yy-mm-dd)",
        "start_time" : "21:25:00",
        "duration" : "00:25:00",
        "running_status" :0,
        "running_status_descr" : "undefined",
        "free_ca_mode" :0,
        "descriptors_loop_length" : 568,
"EIT_descriptors":[

{
        "tag" :77,
        "len" : 40,
        "descr" : "Short event descriptor",
        "short_evt":{
        "language" : "deu",
        "name" : "American Dad!",
        "text" : "Jack ist wieder da"
}
},
{
        "tag" :78,
        "len" : 257,
        "descr" : "Extended event descriptor",
        "ext_evt":{
        "descr_number" : 1 ,
        "descr_total_number" : 2,
        "language" : "deu",
        "items":[],
        "text": "Da Steve ein begeisterter Fahrradfahrer ist, hat er Stan und sich zu einem Vater-Sohn-Rennen angemeldet. Während Steve deshalb vollkommen euphorisch ist und über nichts anderes mehr spricht, hält sich Stans Freude jedoch in Grenzen. Der Grund dafür"
}
},
{
        "tag" :78,
        "len" : 256,
        "descr" : "Extended event descriptor",
        "ext_evt":{
        "descr_number" : 2 ,
        "descr_total_number" : 2,
        "language" : "deu",
        "items":[],
        "text": " liegt angeblich in einem Erlebnis aus Stans Kindheit, welches sein Leben geprägt hat: Er hat als kleiner Junge seinem besten Freund ein paar Tricks auf dem Fahrrad gezeigt, sodass dieser vom Straßenverkehr abgelenkt war und bei einem tragischen U"
}
},
{
        "tag" :84,
        "len" : 4,
        "descr" : "Content descriptor (type of program cf tabl 28 EN 300 468 V1.13.1)",
        "content":{
        "content_nibble_lvl_1" :0,
        "content_nibble_lvl_2" :0,
        "user byte" :0
}
},
{
        "tag" :95,
        "len" : 6,
        "descr" : "Private data specifier descriptor"
},
{
        "tag" :150,
        "len" : 5,
        "descr" : "Unknown descriptor"
}]
},{
        "event_id" :57541,
        "start_time day " : "2016-02-05 (yy-mm-dd)",
        "start_time" : "21:50:00",
        "duration" : "00:25:00",
        "running_status" :0,
        "running_status_descr" : "undefined",
        "free_ca_mode" :0,
        "descriptors_loop_length" : 563,
"EIT_descriptors":[

{
        "tag" :77,
        "len" : 35,
        "descr" : "Short event descriptor",
        "short_evt":{
        "language" : "deu",
        "name" : "American Dad!",
        "text" : "Der Bar-Mizwa"
}
},
{
        "tag" :78,
        "len" : 257,
        "descr" : "Extended event descriptor",
        "ext_evt":{
        "descr_number" : 1 ,
        "descr_total_number" : 2,
        "language" : "deu",
        "items":[],
        "text": "Bei ihrem gemeinsamen Mittagessen in der Shoppingmall treffen Steve und Debbie zufällig auf Steves Freund Snot und dessen jüdischen Freund Etan. Während Steve Etan auf Anhieb unsympathisch  findet, ist Debbie von ihm vollkommen fasziniert, da er se"
}
},
{
        "tag" :78,
        "len" : 256,
        "descr" : "Extended event descriptor",
        "ext_evt":{
        "descr_number" : 2 ,
        "descr_total_number" : 2,
        "language" : "deu",
        "items":[],
        "text": "hr erwachsen wirkt. Als Etan wenige Tage später zu seiner Bar-Mizwa Feier einlädt, trennt sich Debbie von Steve mit der Begründung, sie brauche einen richtigen Mann. Dieser ist daraufhin am Boden zerstört und möchte sich an Etan dafür rächen, dass"
}
},
{
        "tag" :84,
        "len" : 4,
        "descr" : "Content descriptor (type of program cf tabl 28 EN 300 468 V1.13.1)",
        "content":{
        "content_nibble_lvl_1" :0,
        "content_nibble_lvl_2" :0,
        "user byte" :0
}
},
{
        "tag" :95,
        "len" : 6,
        "descr" : "Private data specifier descriptor"
},
{
        "tag" :150,
        "len" : 5,
        "descr" : "Unknown descriptor"
}]
},{
        "event_id" :57585,
        "start_time day " : "2016-02-05 (yy-mm-dd)",
        "start_time" : "22:15:00",
        "duration" : "00:25:00",
        "running_status" :0,
        "running_status_descr" : "undefined",
        "free_ca_mode" :0,
        "descriptors_loop_length" : 567,
"EIT_descriptors":[

{
        "tag" :77,
        "len" : 39,
        "descr" : "Short event descriptor",
        "short_evt":{
        "language" : "deu",
        "name" : "American Dad!",
        "text" : "Die Ersatzehefrau"
}
},
{
        "tag" :78,
        "len" : 257,
        "descr" : "Extended event descriptor",
        "ext_evt":{
        "descr_number" : 1 ,
        "descr_total_number" : 2,
        "language" : "deu",
        "items":[],
        "text": "Francine hat ein ungutes Gefühl, als sich Stan auf eine gefährliche Mission nach Kolumbien begibt. Deshalb versucht sie, sich während seiner Abwesenheit permanent abzulenken, bis auf einmal Stans Chef Bullock vor der Türe steht und ihr mitteilt, da"
}
},
{
        "tag" :78,
        "len" : 256,
        "descr" : "Extended event descriptor",
        "ext_evt":{
        "descr_number" : 2 ,
        "descr_total_number" : 2,
        "language" : "deu",
        "items":[],
        "text": "ss die CIA die Verbindung zu Stan und seinem Kollegen Jim verloren hat. Stan und Jim wurden in Kolumbien verhaftet - doch dank Jims Talent, Frauen zu verführen, gelingt es ihnen, die Wächterinnen zu überlisten und aus dem Gefängnis zu fliehen. Fra"
}
},
{
        "tag" :84,
        "len" : 4,
        "descr" : "Content descriptor (type of program cf tabl 28 EN 300 468 V1.13.1)",
        "content":{
        "content_nibble_lvl_1" :0,
        "content_nibble_lvl_2" :0,
        "user byte" :0
}
},
{
        "tag" :95,
        "len" : 6,
        "descr" : "Private data specifier descriptor"
},
{
        "tag" :150,
        "len" : 5,
        "descr" : "Unknown descriptor"
}]
},{
        "event_id" :36420,
        "start_time day " : "2016-02-05 (yy-mm-dd)",
        "start_time" : "22:40:00",
        "duration" : "00:20:00",
        "running_status" :0,
        "running_status_descr" : "undefined",
        "free_ca_mode" :0,
        "descriptors_loop_length" : 570,
"EIT_descriptors":[

{
        "tag" :77,
        "len" : 42,
        "descr" : "Short event descriptor",
        "short_evt":{
        "language" : "deu",
        "name" : "American Dad!",
        "text" : "Steve allein zu Haus"
}
},
{
        "tag" :78,
        "len" : 257,
        "descr" : "Extended event descriptor",
        "ext_evt":{
        "descr_number" : 1 ,
        "descr_total_number" : 2,
        "language" : "deu",
        "items":[],
        "text": "Obwohl Francine große Bedenken hat, dass Steve bereits alt genug ist, um alleine zu Hause zu bleiben, fliegt sie gemeinsam mit Stan und Hayley zu dem College, das diese in den kommenden Jahren besuchen soll. Zunächst hält sich Steve auch an die ver"
}
},
{
        "tag" :78,
        "len" : 256,
        "descr" : "Extended event descriptor",
        "ext_evt":{
        "descr_number" : 2 ,
        "descr_total_number" : 2,
        "language" : "deu",
        "items":[],
        "text": "einbarten Regeln - bis seine Freunde unverhofft vor der Türe stehen. Steve hat ihnen zwar verboten, das Arbeitszimmer seines Vaters zu betreten, doch schließlich siegt die Neugier der Jungs. Als sie im Arbeitszimmer einen Koffer finden, halten sie"
}
},
{
        "tag" :84,
        "len" : 4,
        "descr" : "Content descriptor (type of program cf tabl 28 EN 300 468 V1.13.1)",
        "content":{
        "content_nibble_lvl_1" :0,
        "content_nibble_lvl_2" :0,
        "user byte" :0
}
},
{
        "tag" :95,
        "len" : 6,
        "descr" : "Private data specifier descriptor"
},
{
        "tag" :150,
        "len" : 5,
        "descr" : "Unknown descriptor"
}]
}

**Comment by braice** [comment]
Hello

Concerning the newlines I think I see the bug it's in the function

int convert_en300468_string(char *string, int max_len, int debug)
 in log.c
I think I should check control character for encoding only at the first
byte and not all the time. I am dropping all ascii non printable control
characters, I should do it but only for the first byte.

To confirm that can you, in eit_show_ext_evt_descr in unicast_EIT.c call
this function with a 1 as last paremeter and give me the full log

I could be linked with the other bug you are seeing, but for this I would
also like to have the full verbose logs;

Thank you

2016-02-05 16:59 GMT-05:00 SourceDoctor notifications@github.com:

**Comment by SourceDoctor** [comment]
Hello,

the confirmation with set the last Argument in "eit_show_ext_evt_descr"
results at me in a compilation Error, maybe you can test it yourself.

For the other bug, i attached you the following files:
- Configfile
- verbose Output of Logging
- eit.json

i hope this helps.
greetings

Am 21.02.2016 um 20:43 schrieb Brice Dubost:

**Comment by braice** [comment]
Hello

I don' t see the attachements,
concerning the other issue (return chars) can you try with this log.c file

Thank you

2016-02-27 17:26 GMT-05:00 SourceDoctor notifications@github.com:

**Comment by SourceDoctor** [comment]
Hello,

i don't see an attached log.c either.

can you send it to me from an other Mail-Account maybe?
So i'll do the Tests for you.
Also i'll send you the Files to this Mail Adresse.

Greetings

Am 06.03.2016 um 23:47 schrieb Brice Dubost:

**Comment by braice** [comment]
Hi

I have pushed a commit which I hope solve the line return issue.
For the other files please send them at mumudvb@braice.net 

You can also open another issue about it, it will be easier to follow.

Thank you

Brice

**Comment by SourceDoctor** [comment]
Hi,

i tried it today to compile but it produces an Error because of a
missing Declaration.
Maybe you forgot something in SourceCode?

root@streamserver:~/MuMuDVB# autoreconf -i -f
configure.ac:13: installing './compile'
configure.ac:27: installing './config.guess'
configure.ac:27: installing './config.sub'
configure.ac:7: installing './install-sh'
configure.ac:7: installing './missing'
doc/Makefile.am:9: warning: '%'-style pattern rules are a GNU make extension
src/Makefile.am: installing './depcomp'
root@streamserver:~/MuMuDVB# ./configure
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking for style of include used by make... GNU
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... gcc3
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking linux/dvb/dmx.h usability... yes
checking linux/dvb/dmx.h presence... yes
checking for linux/dvb/dmx.h... yes
checking for gcc... (cached) gcc
checking whether we are using the GNU C compiler... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to accept ISO C89... (cached) none needed
checking whether gcc understands -c and -o together... (cached) yes
checking dependency style of gcc... (cached) gcc3
checking whether make sets $(MAKE)... (cached) yes
checking for gcc option to accept ISO C99... -std=gnu99
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking for ld used by gcc -std=gnu99... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for shared library run path origin... done
checking for iconv... yes
checking for working iconv... yes
checking for iconv declaration...
         extern size_t iconv (iconv_t cd, char \* *inbuf, size_t
*inbytesleft, char \* *outbuf, size_t *outbytesleft);
checking for dvbdemux_set_section_filter in -ldvbapi... yes
checking for atsc_text_segment_decode in -lucsi... yes
checking for pthread_create in -lpthread... yes
checking for clock_gettime in -lrt... yes
checking for log2l in -lm... yes
checking for atsc_text_segment_decode in -lucsi... (cached) yes
checking for en50221_tl_create in -ldvben50221... yes
checking for dvbcsa_bs_decrypt in -ldvbcsa... yes
checking for sys/types.h... (cached) yes
checking for netinet/in.h... yes
checking for arpa/nameser.h... yes
checking for netdb.h... yes
checking for resolv.h... yes
checking arpa/inet.h usability... yes
checking arpa/inet.h presence... yes
checking for arpa/inet.h... yes
checking fcntl.h usability... yes
checking fcntl.h presence... yes
checking for fcntl.h... yes
checking for netdb.h... (cached) yes
checking for netinet/in.h... (cached) yes
checking for stdint.h... (cached) yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking sys/ioctl.h usability... yes
checking sys/ioctl.h presence... yes
checking for sys/ioctl.h... yes
checking sys/socket.h usability... yes
checking sys/socket.h presence... yes
checking for sys/socket.h... yes
checking sys/time.h usability... yes
checking sys/time.h presence... yes
checking for sys/time.h... yes
checking syslog.h usability... yes
checking syslog.h presence... yes
checking for syslog.h... yes
checking for unistd.h... (cached) yes
checking values.h usability... yes
checking values.h presence... yes
checking for values.h... yes
checking for int32_t... yes
checking for size_t... yes
checking for uint16_t... yes
checking for uint32_t... yes
checking for uint8_t... yes
checking for alarm... yes
checking for gettimeofday... yes
checking for inet_ntoa... yes
checking for memset... yes
checking for socket... yes
checking for strerror... yes
checking for strstr... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating doc/Makefile
config.status: creating src/Makefile
config.status: creating src/config.h
config.status: executing depfiles commands

---

MuMuDVB configure results:

Features

Build with CAM support:                             yes
Build with SCAM support:                            yes
Build with ATSC long names support:                 yes
Build with compatibility for android:                no

Debugging

Build with code coverage debug symbols               no
Build with debugging DUMA library                    no
root@streamserver:~/MuMuDVB# make
Making all in src
make[1]: Entering directory `/root/MuMuDVB/src'
make  all-am
make[2]: Entering directory`/root/MuMuDVB/src'
gcc -std=gnu99 -DHAVE_CONFIG_H -I.    -Wall -Wextra -g -O2 -MT
autoconf.o -MD -MP -MF .deps/autoconf.Tpo -c -o autoconf.o autoconf.c
gcc -std=gnu99 -DHAVE_CONFIG_H -I.    -Wall -Wextra -g -O2 -MT crc32.o
-MD -MP -MF .deps/crc32.Tpo -c -o crc32.o crc32.c
gcc -std=gnu99 -DHAVE_CONFIG_H -I.    -Wall -Wextra -g -O2 -MT log.o -MD
-MP -MF .deps/log.Tpo -c -o log.o log.c
gcc -std=gnu99 -DHAVE_CONFIG_H -I.    -Wall -Wextra -g -O2 -MT
multicast.o -MD -MP -MF .deps/multicast.Tpo -c -o multicast.o multicast.c
gcc -std=gnu99 -DHAVE_CONFIG_H -I.    -Wall -Wextra -g -O2 -MT dvb.o -MD
-MP -MF .deps/dvb.Tpo -c -o dvb.o dvb.c
gcc -std=gnu99 -DHAVE_CONFIG_H -I.    -Wall -Wextra -g -O2 -MT mumudvb.o
-MD -MP -MF .deps/mumudvb.Tpo -c -o mumudvb.o mumudvb.c
gcc -std=gnu99 -DHAVE_CONFIG_H -I.    -Wall -Wextra -g -O2 -MT
mumudvb_mon.o -MD -MP -MF .deps/mumudvb_mon.Tpo -c -o mumudvb_mon.o
mumudvb_mon.c
gcc -std=gnu99 -DHAVE_CONFIG_H -I.    -Wall -Wextra -g -O2 -MT
mumudvb_common.o -MD -MP -MF .deps/mumudvb_common.Tpo -c -o
mumudvb_common.o mumudvb_common.c
gcc -std=gnu99 -DHAVE_CONFIG_H -I.    -Wall -Wextra -g -O2 -MT network.o
-MD -MP -MF .deps/network.Tpo -c -o network.o network.c
mv -f .deps/crc32.Tpo .deps/crc32.Po
gcc -std=gnu99 -DHAVE_CONFIG_H -I.    -Wall -Wextra -g -O2 -MT
rewrite_pat.o -MD -MP -MF .deps/rewrite_pat.Tpo -c -o rewrite_pat.o
rewrite_pat.c
log.c: In function ‘convert_en300468_string’:
log.c:1164:2: error: ‘dest’ undeclared (first use in this function)
  dest=string;
  ^
log.c:1164:2: note: each undeclared identifier is reported only once for
each function it appears in
make[2]: **\* [log.o] Error 1
make[2]: **\* Waiting for unfinished jobs....
mv -f .deps/multicast.Tpo .deps/multicast.Po
mv -f .deps/network.Tpo .deps/network.Po
mv -f .deps/rewrite_pat.Tpo .deps/rewrite_pat.Po
mv -f .deps/mumudvb_common.Tpo .deps/mumudvb_common.Po
mv -f .deps/dvb.Tpo .deps/dvb.Po
mv -f .deps/autoconf.Tpo .deps/autoconf.Po
mv -f .deps/mumudvb_mon.Tpo .deps/mumudvb_mon.Po
mv -f .deps/mumudvb.Tpo .deps/mumudvb.Po
make[2]: Leaving directory `/root/MuMuDVB/src'
make[1]: *** [all] Error 2
make[1]: Leaving directory`/root/MuMuDVB/src'
make: **\* [all-recursive] Error 1

Am 07.03.2016 um 00:12 schrieb Brice Dubost:

**Comment by braice** [comment]
Corrected, sorry for that

2016-03-07 15:39 GMT-05:00 SourceDoctor notifications@github.com:

**Comment by braice** [comment]
Hi,

I looked a bit more into your stuff

In the event I selected below, if I look to the length of the descriptors
which are decoded I have

39 + 256 + 257 + 4 + 6 + 5 = 567

Containing the short and the two long descriptions so there is no obvious
miss from MuMuDVB as all the data announced is decoded

Maybe they just take the very long description (
http://www.teleboy.ch/fr/programme/RTLNitroHD/2955989/american-dad in this
case) and copy a part into the descriptor.

The descriptors not decoded are
{
"tag" :95,
"len" : 6,
"descr" : "Private data specifier descriptor"
},

--> it says who have put the private data (eg page 24 of
http://www.broadcasting.ru/pdf-standard-specifications/multiplexing/dvb-si/dtr10116-v1-2-1-010129.pdf
)

This one is private data and is not in the norm but it is only 5 bytes
{
"tag" :150,
"len" : 5,
"descr" : "Unknown descriptor"
}]

So in conclusion I am afraid that it is more a provider bug than a MuMuDVB
bug.

Sorry

Brice

2016-02-05 16:59 GMT-05:00 SourceDoctor notifications@github.com:

> Hi Brice,
> 
> i think i've maybe found another Bug in EPG Handling:
> 
> On some Transponders like this one:
> Astra 19.2E
> freq=12187
> pol=h
> srate=27500
> EPG seems to be buggy. It has maximal 2 extended EPG Parts and end's in
> middle of a sentence.
> 
> Other Transponders like:
> Astra 19.2E
> freq=12544
> pol=h
> srate=22000
> With same MuMuDVB Configuration is running well and showing extended EPG
> complete, even if there are 4 or more extended EPG Parts.
> 
> i attached you a matching Part of the JSON of the Problem.
> I hope you'll find the source of the Problem.
> 
> },{
> "event_id" :57585,
> "start_time day " : "2016-02-05 (yy-mm-dd)",
> "start_time" : "22:15:00",
> "duration" : "00:25:00",
> "running_status" :0,
> "running_status_descr" : "undefined",
> "free_ca_mode" :0,
> "descriptors_loop_length" : 567,
> "EIT_descriptors":[
> 
> {
> "tag" :77,
> "len" : 39,
> "descr" : "Short event descriptor",
> "short_evt":{
> "language" : "deu",
> "name" : "American Dad!",
> "text" : "Die Ersatzehefrau"
> }
> },
> {
> "tag" :78,
> "len" : 257,
> "descr" : "Extended event descriptor",
> "ext_evt":{
> "descr_number" : 1 ,
> "descr_total_number" : 2,
> "language" : "deu",
> "items":[],
> "text": "Francine hat ein ungutes Gefühl, als sich Stan auf eine
> gefährliche Mission nach Kolumbien begibt. Deshalb versucht sie, sich
> während seiner Abwesenheit permanent abzulenken, bis auf einmal Stans Chef
> Bullock vor der Türe steht und ihr mitteilt, da"
> }
> },
> {
> "tag" :78,
> "len" : 256,
> "descr" : "Extended event descriptor",
> "ext_evt":{
> "descr_number" : 2 ,
> "descr_total_number" : 2,
> "language" : "deu",
> "items":[],
> "text": "ss die CIA die Verbindung zu Stan und seinem Kollegen Jim
> verloren hat. Stan und Jim wurden in Kolumbien verhaftet - doch dank Jims
> Talent, Frauen zu verführen, gelingt es ihnen, die Wächterinnen zu
> überlisten und aus dem Gefängnis zu fliehen. Fra"
> }
> },
> {
> "tag" :84,
> "len" : 4,
> "descr" : "Content descriptor (type of program cf tabl 28 EN 300 468
> V1.13.1)",
> "content":{
> "content_nibble_lvl_1" :0,
> "content_nibble_lvl_2" :0,
> "user byte" :0
> }
> },
> {
> "tag" :95,
> "len" : 6,
> "descr" : "Private data specifier descriptor"
> },
> {
> "tag" :150,
> "len" : 5,
> "descr" : "Unknown descriptor"
> }]

**Comment by SourceDoctor** [comment]
Hi Brice,

it's me again.

After a long time now i had time to check the EIT fix you wrote for
transporting all received Characters in EPG.
I took the newest mumudvb2-Branch.

On trying to fetch EIT.json from MuMuDVB-API mumudvb crashes immediatly.

i hope the attached log helps.

with regards.
   Thomas

Deb2:  TS:  General information PID 18 adaptation_field_control 1 payload_unit_start_indicator 0 continuity_counter 3
Deb2:  TS:  Continuing packet, data left 184
Deb2:  TS:  Continuing a packet PID 18 cc 3 len 2575 expected 2583
Deb2:  TS:  General information PID 18 adaptation_field_control 1 payload_unit_start_indicator 0 continuity_counter 4
Deb2:  TS:  Continuing packet, data left 184
Deb2:  TS:  Continuing a packet PID 18 cc 4 len 2583 expected 2583
Deb2:  TS:  New full packet len 2583. There's now 1 full packet
Deb2:  TS:  Full packet left: 1, we copy length 2583
Deb2:  TS:  General information PID 18 adaptation_field_control 1 payload_unit_start_indicator 1 continuity_counter 5
Deb2:  TS:  Starting a packet PID 18 cc 5 len 183 expected len 460
Deb2:  TS:  First bytes  0x4e 0xf1 0xc9 0x2f  0x3a 0xd3 0x01 0x01
Deb2:  TS:  Struct data  table_id 0x4e section_syntax_indicator 0x01 section_length 0x1c9 transport_stream_id 0x2f3a version_number 0x09 current_next_indicator                                        0x01 last_section_number 0x01
Deb2:  Unicast :  New client
Deb2:  Unicast :  New connection from 127.0.0.1:48800 to 127.0.0.1:19999
Deb2:  Unicast :  We create a client associated with the socket 64
Deb2:  Unicast :  first client
Deb2:  Unicast :  Actual SO_SNDBUF size : 2626560
Deb2:  Unicast :  Number of clients : 1
Deb2:  Unicast :  New message for socket 64
Deb2:  Unicast :  beginning of buffer GET /m
Deb2:  Unicast :  beginning of buffer 71 69 84 32 47 109
Deb2:  Unicast :  We received 100, buffer len 100 new buffer pos 100
Deb2:  TS:  General information PID 18 adaptation_field_control 1 payload_unit_start_indicator 0 continuity_counter 6
Deb2:  TS:  Continuing packet, data left 184
Deb2:  TS:  Continuing a packet PID 18 cc 6 len 367 expected 460
Deb2:  Unicast :  New message for socket 64
Deb2:  Unicast :  We received 33, buffer len 200 new buffer pos 133
Deb2:  Unicast :  End of HTTP request, we parse it
Deb0:  Unicast :  EIT Json
**\* Error in `./mumudvb': free(): invalid next size (fast): 0x000000000080d260 \*                                       **
Aborted (core dumped)

**Comment by braice** [comment]
Hi

Can you do a backtrace with gdb?

**Comment by SourceDoctor** [comment]
no problem,

but i'm not familiar with gdb,

could you give me the command how to do it in correct form for you?

Am 29.04.2016 um 21:14 schrieb Brice Dubost:

> Hi
> 
> Can you do a backtrace with gdb?

**Comment by braice** [comment]
Type
Gdb ./src/mumudvb
Then
Run "your usual command line arguments"
Then when it crashes
Bt

Thank you

**Comment by SourceDoctor** [comment]
Hi Brice,

i attached you the backtrace.

hope it helps

with regards
Thomas

Am 30.04.2016 um 03:00 schrieb Brice Dubost:

> Type
> Gdb ./src/mumudvb
> Then
> Run "your usual command line arguments"
> Then when it crashes
> Bt
> 
> Thank you
> On Apr 29, 2016 15:23, "SourceDoctor" notifications@github.com wrote:
> 
> > no problem,
> > 
> > but i'm not familiar with gdb,
> > 
> > could you give me the command how to do it in correct form for you?
> > 
> > Am 29.04.2016 um 21:14 schrieb Brice Dubost:
> > 
> > > Hi
> > > 
> > > Can you do a backtrace with gdb?
> > > On Apr 29, 2016 07:53, "SourceDoctor" notifications@github.com
> > > wrote:
> > > 
> > > > Hi Brice,
> > > > 
> > > > it's me again.
> > > > 
> > > > After a long time now i had time to check the EIT fix you wrote for
> > > > transporting all received Characters in EPG.
> > > > I took the newest mumudvb2-Branch.
> > > > 
> > > > On trying to fetch EIT.json from MuMuDVB-API mumudvb crashes
> > > > immediatly.
> > > > 
> > > > i hope the attached log helps.
> > > > 
> > > > with regards.
> > > > Thomas
> > > > 
> > > > Deb2: TS: General information PID 18 adaptation_field_control 1
> > > > payload_unit_start_indicator 0 continuity_counter 3
> > > > Deb2: TS: Continuing packet, data left 184
> > > > Deb2: TS: Continuing a packet PID 18 cc 3 len 2575 expected 2583
> > > > Deb2: TS: General information PID 18 adaptation_field_control 1
> > > > payload_unit_start_indicator 0 continuity_counter 4
> > > > Deb2: TS: Continuing packet, data left 184
> > > > Deb2: TS: Continuing a packet PID 18 cc 4 len 2583 expected 2583
> > > > Deb2: TS: New full packet len 2583. There's now 1 full packet
> > > > Deb2: TS: Full packet left: 1, we copy length 2583
> > > > Deb2: TS: General information PID 18 adaptation_field_control 1
> > > > payload_unit_start_indicator 1 continuity_counter 5
> > > > Deb2: TS: Starting a packet PID 18 cc 5 len 183 expected len 460
> > > > Deb2: TS: First bytes 0x4e 0xf1 0xc9 0x2f 0x3a 0xd3 0x01 0x01
> > > > Deb2: TS: Struct data table_id 0x4e section_syntax_indicator 0x01
> > > > section_length 0x1c9 transport_stream_id 0x2f3a version_number 0x09
> > > > current_next_indicator 0x01 last_section_number 0x01
> > > > Deb2: Unicast : New client
> > > > Deb2: Unicast : New connection from 127.0.0.1:48800 to
> > > > 127.0.0.1:19999
> > > > Deb2: Unicast : We create a client associated with the socket 64
> > > > Deb2: Unicast : first client
> > > > Deb2: Unicast : Actual SO_SNDBUF size : 2626560
> > > > Deb2: Unicast : Number of clients : 1
> > > > Deb2: Unicast : New message for socket 64
> > > > Deb2: Unicast : beginning of buffer GET /m
> > > > Deb2: Unicast : beginning of buffer 71 69 84 32 47 109
> > > > Deb2: Unicast : We received 100, buffer len 100 new buffer pos 100
> > > > Deb2: TS: General information PID 18 adaptation_field_control 1
> > > > payload_unit_start_indicator 0 continuity_counter 6
> > > > Deb2: TS: Continuing packet, data left 184
> > > > Deb2: TS: Continuing a packet PID 18 cc 6 len 367 expected 460
> > > > Deb2: Unicast : New message for socket 64
> > > > Deb2: Unicast : We received 33, buffer len 200 new buffer pos 133
> > > > Deb2: Unicast : End of HTTP request, we parse it
> > > > Deb0: Unicast : EIT Json
> > > > **\* Error in `./mumudvb': free(): invalid next size (fast):
> > > > 0x000000000080d260 \* **
> > > > Aborted (core dumped)
> > > > 
> > > > —
> > > > You are receiving this because you modified the open/close state.
> > > > Reply to this email directly or view it on GitHub
> > > > 
> > > > https://github.com/braice/MuMuDVB/issues/100#issuecomment-215692360
> > > 
> > > —
> > > You are receiving this because you authored the thread.
> > > Reply to this email directly or view it on GitHub
> > > https://github.com/braice/MuMuDVB/issues/100#issuecomment-215852325
> > 
> > —
> > You are receiving this because you modified the open/close state.
> > Reply to this email directly or view it on GitHub
> > https://github.com/braice/MuMuDVB/issues/100#issuecomment-215854225
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/braice/MuMuDVB/issues/100#issuecomment-215918438

Deb2:  TS:  General information PID 18 adaptation_field_control 1 payload_unit_start_indicator 0 continuity_counter 4
Deb2:  TS:  Continuing packet, data left 184
Deb2:  TS:  Continuing a packet PID 18 cc 4 len 367 expected 1225
Deb2:  TS:  General information PID 18 adaptation_field_control 1 payload_unit_start_indicator 0 continuity_counter 5
Deb2:  TS:  Continuing packet, data left 184
Deb2:  TS:  Continuing a packet PID 18 cc 5 len 551 expected 1225
Deb2:  TS:  General information PID 18 adaptation_field_control 1 payload_unit_start_indicator 0 continuity_counter 6
Deb2:  TS:  Continuing packet, data left 184
Deb2:  TS:  Continuing a packet PID 18 cc 6 len 735 expected 1225
Deb2:  TS:  General information PID 18 adaptation_field_control 1 payload_unit_start_indicator 0 continuity_counter 7
Deb2:  TS:  Continuing packet, data left 184
Deb2:  TS:  Continuing a packet PID 18 cc 7 len 919 expected 1225
Deb2:  TS:  General information PID 18 adaptation_field_control 1 payload_unit_start_indicator 0 continuity_counter 8
Deb2:  TS:  Continuing packet, data left 184
Deb2:  TS:  Continuing a packet PID 18 cc 8 len 1103 expected 1225
Deb2:  TS:  General information PID 18 adaptation_field_control 1 payload_unit_start_indicator 0 continuity_counter 9
Deb2:  TS:  Continuing packet, data left 184
Deb2:  TS:  Continuing a packet PID 18 cc 9 len 1225 expected 1225
Deb2:  TS:  New full packet len 1225. There's now 1 full packet
Deb2:  TS:  Full packet left: 1, we copy length 1225
Deb2:  TS:  General information PID 18 adaptation_field_control 1 payload_unit_start_indicator 1 continuity_counter 10
Deb2:  TS:  Starting a packet PID 18 cc 10 len 183 expected len 949
Deb2:  TS:  First bytes  0x51 0xf3 0xb2 0x2e  0xf4 0xe1 0xb8 0xf8
Deb2:  TS:  Struct data  table_id 0x51 section_syntax_indicator 0x01 section_length 0x3b2 transport_stream_id 0x2ef4 version_number 0x10 current_next_indicator 0x01 last_section_number 0xf8
Deb2:  TS:  General information PID 18 adaptation_field_control 1 payload_unit_start_indicator 0 continuity_counter 11
Deb2:  TS:  Continuing packet, data left 184
Deb2:  TS:  Continuing a packet PID 18 cc 11 len 367 expected 949
Deb2:  TS:  General information PID 18 adaptation_field_control 1 payload_unit_start_indicator 0 continuity_counter 12
Deb2:  TS:  Continuing packet, data left 184
Deb2:  TS:  Continuing a packet PID 18 cc 12 len 551 expected 949
Deb2:  TS:  General information PID 18 adaptation_field_control 1 payload_unit_start_indicator 0 continuity_counter 13
Deb2:  TS:  Continuing packet, data left 184
Deb2:  TS:  Continuing a packet PID 18 cc 13 len 735 expected 949
Deb2:  Unicast :  New client
Deb2:  Unicast :  New connection from 127.0.0.1:62343 to 127.0.0.1:19999
Deb2:  Unicast :  We create a client associated with the socket 64
Deb2:  Unicast :  first client
Deb2:  Unicast :  Actual SO_SNDBUF size : 2626560
Deb2:  Unicast :  Number of clients : 1
Deb2:  Unicast :  New message for socket 64
Deb2:  Unicast :  beginning of buffer GET /m
Deb2:  Unicast :  beginning of buffer 71 69 84 32 47 109
Deb2:  Unicast :  We received 94, buffer len 100 new buffer pos 94
Deb2:  Unicast :  End of HTTP request, we parse it
Deb0:  Unicast :  EIT Json
**\* Error in `/root/mumudvb': free(): invalid next size (fast): 0x000000000063e540 ***

Program received signal SIGABRT, Aborted.
0x00007ffff6adfcc9 in raise () from /lib/x86_64-linux-gnu/libc.so.6
(gdb) bt
#0  0x00007ffff6adfcc9 in raise () from /lib/x86_64-linux-gnu/libc.so.6
#1  0x00007ffff6ae30d8 in abort () from /lib/x86_64-linux-gnu/libc.so.6
#2  0x00007ffff6b1c394 in ?? () from /lib/x86_64-linux-gnu/libc.so.6
#3  0x00007ffff6b2866e in ?? () from /lib/x86_64-linux-gnu/libc.so.6
#4  0x00000000004088ec in convert_en300468_string (
    string=string@entry=0x7ffffff2b2c0 "uf und davon - Mein Auslandstagebuch", max_len=max_len@entry=4096,
    debug=debug@entry=0) at log.c:1170
#5  0x0000000000418e4c in eit_show_short_evt_descr (
    buf=buf@entry=0x77f759 "M+deu&\005Auf und davon - Mein Auslandstagebuch", reply=reply@entry=0xe591e0)
    at unicast_EIT.c:476
#6  0x0000000000419579 in eit_display_descriptor (
    buf=buf@entry=0x77f759 "M+deu&\005Auf und davon - Mein Auslandstagebuch", descriptors_loop_len=407,
    reply=reply@entry=0xe591e0) at unicast_EIT.c:282
#7  0x000000000041997b in eit_display_contents (full_eit=full_eit@entry=0x77f2e0,
    reply=reply@entry=0xe591e0) at unicast_EIT.c:239
#8  0x0000000000419a09 in unicast_send_EIT_section (eit_section=0x77f2e0, num=num@entry=80,
    reply=reply@entry=0xe591e0) at unicast_EIT.c:37
#9  0x000000000041cab3 in unicast_send_EIT (eit_packets=eit_packets@entry=0x6844c0, Socket=64)
    at unicast_monit.c:259
#10 0x00000000004178eb in unicast_handle_message (unicast_vars=unicast_vars@entry=0x7ffffff36150,
    client=0xe59330, channels=channels@entry=0x7ffffff38548,
    number_of_channels=number_of_channels@entry=12, strengthparams=strengthparams@entry=0x7ffffff2d9b0,
    auto_p=auto_p@entry=0x7ffffff36de0, cam_p=cam_p@entry=0x7ffffff364d0,
    scam_vars=scam_vars@entry=0x7ffffff36080, eit_packets=eit_packets@entry=0x6844c0) at unicast_http.c:818
#11 0x0000000000417c5f in unicast_handle_fd_event (unicast_vars=unicast_vars@entry=0x7ffffff36150,
    channels=channels@entry=0x7ffffff38548, number_of_channels=12,
    strengthparams=strengthparams@entry=0x7ffffff2d9b0, auto_p=auto_p@entry=0x7ffffff36de0,
    cam_p=cam_p@entry=0x7ffffff364d0, scam_vars=scam_vars@entry=0x7ffffff36080, eit_packets=0x6844c0)
    at unicast_http.c:391
#12 0x0000000000406005 in main (argc=<optimized out>, argv=<optimized out>) at mumudvb.c:1289
(gdb)

**Comment by braice** [comment]
Hello

Can you comment line 1170 of log.c and check again,

if it works can you check with valgrind for memory leaks ? (just run
mumudvb with valgrind before in the command line)

Thank you

2016-04-30 19:47 GMT-04:00 SourceDoctor notifications@github.com:

> Hi Brice,
> 
> i attached you the backtrace.
> 
> hope it helps
> 
> with regards
> Thomas
> 
> Am 30.04.2016 um 03:00 schrieb Brice Dubost:
> 
> > Type
> > Gdb ./src/mumudvb
> > Then
> > Run "your usual command line arguments"
> > Then when it crashes
> > Bt
> > 
> > Thank you

**Comment by SourceDoctor** [comment]
commenting out this line didn't change anything

Am 02.05.2016 um 01:29 schrieb Brice Dubost:

**Comment by braice** [comment]
You commented the line free(...) ?

If yes can you send me the new backtrace?

Thank you

**Comment by SourceDoctor** [comment]
correct, i commented out the "free" - Line

Backtrace is attached.

Am 03.05.2016 um 03:22 schrieb Brice Dubost:

> You commented the line free(...) ?
> 
> If yes can you send me the new backtrace?
> 
> Thank you
> On May 2, 2016 20:10, "SourceDoctor" notifications@github.com wrote:
> 
> > commenting out this line didn't change anything
> > 
> > Am 02.05.2016 um 01:29 schrieb Brice Dubost:
> > 
> > > Hello
> > > 
> > > Can you comment line 1170 of log.c and check again,
> > > 
> > > if it works can you check with valgrind for memory leaks ? (just run
> > > mumudvb with valgrind before in the command line)
> > > 
> > > Thank you
> > > 
> > > 2016-04-30 19:47 GMT-04:00 SourceDoctor notifications@github.com:
> > > 
> > > > Hi Brice,
> > > > 
> > > > i attached you the backtrace.
> > > > 
> > > > hope it helps
> > > > 
> > > > with regards
> > > > Thomas
> > > > 
> > > > Am 30.04.2016 um 03:00 schrieb Brice Dubost:
> > > > 
> > > > > Type
> > > > > Gdb ./src/mumudvb
> > > > > Then
> > > > > Run "your usual command line arguments"
> > > > > Then when it crashes
> > > > > Bt
> > > > > 
> > > > > Thank you
> > > > > On Apr 29, 2016 15:23, "SourceDoctor" notifications@github.com
> > > > > wrote:
> > > > > 
> > > > > > no problem,
> > > > > > 
> > > > > > but i'm not familiar with gdb,
> > > > > > 
> > > > > > could you give me the command how to do it in correct form
> > > > > > for you?
> > > > > > 
> > > > > > Am 29.04.2016 um 21:14 schrieb Brice Dubost:
> > > > > > 
> > > > > > > Hi
> > > > > > > 
> > > > > > > Can you do a backtrace with gdb?
> > > > > > > On Apr 29, 2016 07:53, "SourceDoctor"
> > > > > > > notifications@github.com
> > > > > > > wrote:
> > > > > > > 
> > > > > > > > Hi Brice,
> > > > > > > > 
> > > > > > > > it's me again.
> > > > > > > > 
> > > > > > > > After a long time now i had time to check the EIT fix you
> > > > > > > > wrote for
> > > > > > > > transporting all received Characters in EPG.
> > > > > > > > I took the newest mumudvb2-Branch.
> > > > > > > > 
> > > > > > > > On trying to fetch EIT.json from MuMuDVB-API mumudvb crashes
> > > > > > > > immediatly.
> > > > > > > > 
> > > > > > > > i hope the attached log helps.
> > > > > > > > 
> > > > > > > > with regards.
> > > > > > > > Thomas
> > > > > > > > 
> > > > > > > > Deb2: TS: General information PID 18
> > > > > > > > adaptation_field_control 1
> > > > > > > > payload_unit_start_indicator 0 continuity_counter 3
> > > > > > > > Deb2: TS: Continuing packet, data left 184
> > > > > > > > Deb2: TS: Continuing a packet PID 18 cc 3 len 2575 expected
> > > > > > > > 2583
> > > > > > > > Deb2: TS: General information PID 18
> > > > > > > > adaptation_field_control 1
> > > > > > > > payload_unit_start_indicator 0 continuity_counter 4
> > > > > > > > Deb2: TS: Continuing packet, data left 184
> > > > > > > > Deb2: TS: Continuing a packet PID 18 cc 4 len 2583 expected
> > > > > > > > 2583
> > > > > > > > Deb2: TS: New full packet len 2583. There's now 1 full
> > > > > > > > packet
> > > > > > > > Deb2: TS: Full packet left: 1, we copy length 2583
> > > > > > > > Deb2: TS: General information PID 18
> > > > > > > > adaptation_field_control 1
> > > > > > > > payload_unit_start_indicator 1 continuity_counter 5
> > > > > > > > Deb2: TS: Starting a packet PID 18 cc 5 len 183 expected len
> > > > > > > > 460
> > > > > > > > Deb2: TS: First bytes 0x4e 0xf1 0xc9 0x2f 0x3a 0xd3 0x01
> > > > > > > > 0x01
> > > > > > > > Deb2: TS: Struct data table_id 0x4e section_syntax_indicator
> > > > > > > > 0x01
> > > > > > > > section_length 0x1c9 transport_stream_id 0x2f3a
> > > > > > > > version_number 0x09
> > > > > > > > current_next_indicator 0x01 last_section_number 0x01
> > > > > > > > Deb2: Unicast : New client
> > > > > > > > Deb2: Unicast : New connection from 127.0.0.1:48800 to
> > > > > > > > 127.0.0.1:19999
> > > > > > > > Deb2: Unicast : We create a client associated with the
> > > > > > > > socket
> > > > > > > > 64
> > > > > > > > Deb2: Unicast : first client
> > > > > > > > Deb2: Unicast : Actual SO_SNDBUF size : 2626560
> > > > > > > > Deb2: Unicast : Number of clients : 1
> > > > > > > > Deb2: Unicast : New message for socket 64
> > > > > > > > Deb2: Unicast : beginning of buffer GET /m
> > > > > > > > Deb2: Unicast : beginning of buffer 71 69 84 32 47 109
> > > > > > > > Deb2: Unicast : We received 100, buffer len 100 new buffer
> > > > > > > > pos 100
> > > > > > > > Deb2: TS: General information PID 18
> > > > > > > > adaptation_field_control 1
> > > > > > > > payload_unit_start_indicator 0 continuity_counter 6
> > > > > > > > Deb2: TS: Continuing packet, data left 184
> > > > > > > > Deb2: TS: Continuing a packet PID 18 cc 6 len 367
> > > > > > > > expected 460
> > > > > > > > Deb2: Unicast : New message for socket 64
> > > > > > > > Deb2: Unicast : We received 33, buffer len 200 new buffer
> > > > > > > > pos 133
> > > > > > > > Deb2: Unicast : End of HTTP request, we parse it
> > > > > > > > Deb0: Unicast : EIT Json
> > > > > > > > **\* Error in `./mumudvb': free(): invalid next size (fast):
> > > > > > > > 0x000000000080d260 \* **
> > > > > > > > Aborted (core dumped)
> > > > > > > > 
> > > > > > > > —
> > > > > > > > You are receiving this because you modified the open/close
> > > > > > > > state.
> > > > > > > > Reply to this email directly or view it on GitHub
> > > > > 
> > > > > <https://github.com/braice/MuMuDVB/issues/100#issuecomment-215692360
> > > > > 
> > > > > > > —
> > > > > > > You are receiving this because you authored the thread.
> > > > > > > Reply to this email directly or view it on GitHub
> > > > > > > 
> > > > > > > <https://github.com/braice/MuMuDVB/issues/100#issuecomment-215852325
> > > > > > 
> > > > > > —
> > > > > > You are receiving this because you modified the open/close
> > > > > > state.
> > > > > > Reply to this email directly or view it on GitHub
> > > > > > 
> > > > > > https://github.com/braice/MuMuDVB/issues/100#issuecomment-215854225
> > > > > 
> > > > > —
> > > > > You are receiving this because you authored the thread.
> > > > > Reply to this email directly or view it on GitHub
> > > > > 
> > > > > <https://github.com/braice/MuMuDVB/issues/100#issuecomment-215918438
> > > > 
> > > > Deb2: TS: General information PID 18 adaptation_field_control 1
> > > > payload_unit_start_indicator 0 continuity_counter 4
> > > > Deb2: TS: Continuing packet, data left 184
> > > > Deb2: TS: Continuing a packet PID 18 cc 4 len 367 expected 1225
> > > > Deb2: TS: General information PID 18 adaptation_field_control 1
> > > > payload_unit_start_indicator 0 continuity_counter 5
> > > > Deb2: TS: Continuing packet, data left 184
> > > > Deb2: TS: Continuing a packet PID 18 cc 5 len 551 expected 1225
> > > > Deb2: TS: General information PID 18 adaptation_field_control 1
> > > > payload_unit_start_indicator 0 continuity_counter 6
> > > > Deb2: TS: Continuing packet, data left 184
> > > > Deb2: TS: Continuing a packet PID 18 cc 6 len 735 expected 1225
> > > > Deb2: TS: General information PID 18 adaptation_field_control 1
> > > > payload_unit_start_indicator 0 continuity_counter 7
> > > > Deb2: TS: Continuing packet, data left 184
> > > > Deb2: TS: Continuing a packet PID 18 cc 7 len 919 expected 1225
> > > > Deb2: TS: General information PID 18 adaptation_field_control 1
> > > > payload_unit_start_indicator 0 continuity_counter 8
> > > > Deb2: TS: Continuing packet, data left 184
> > > > Deb2: TS: Continuing a packet PID 18 cc 8 len 1103 expected 1225
> > > > Deb2: TS: General information PID 18 adaptation_field_control 1
> > > > payload_unit_start_indicator 0 continuity_counter 9
> > > > Deb2: TS: Continuing packet, data left 184
> > > > Deb2: TS: Continuing a packet PID 18 cc 9 len 1225 expected 1225
> > > > Deb2: TS: New full packet len 1225. There's now 1 full packet
> > > > Deb2: TS: Full packet left: 1, we copy length 1225
> > > > Deb2: TS: General information PID 18 adaptation_field_control 1
> > > > payload_unit_start_indicator 1 continuity_counter 10
> > > > Deb2: TS: Starting a packet PID 18 cc 10 len 183 expected len 949
> > > > Deb2: TS: First bytes 0x51 0xf3 0xb2 0x2e 0xf4 0xe1 0xb8 0xf8
> > > > Deb2: TS: Struct data table_id 0x51 section_syntax_indicator 0x01
> > > > section_length 0x3b2 transport_stream_id 0x2ef4 version_number 0x10
> > > > current_next_indicator 0x01 last_section_number 0xf8
> > > > Deb2: TS: General information PID 18 adaptation_field_control 1
> > > > payload_unit_start_indicator 0 continuity_counter 11
> > > > Deb2: TS: Continuing packet, data left 184
> > > > Deb2: TS: Continuing a packet PID 18 cc 11 len 367 expected 949
> > > > Deb2: TS: General information PID 18 adaptation_field_control 1
> > > > payload_unit_start_indicator 0 continuity_counter 12
> > > > Deb2: TS: Continuing packet, data left 184
> > > > Deb2: TS: Continuing a packet PID 18 cc 12 len 551 expected 949
> > > > Deb2: TS: General information PID 18 adaptation_field_control 1
> > > > payload_unit_start_indicator 0 continuity_counter 13
> > > > Deb2: TS: Continuing packet, data left 184
> > > > Deb2: TS: Continuing a packet PID 18 cc 13 len 735 expected 949
> > > > Deb2: Unicast : New client
> > > > Deb2: Unicast : New connection from 127.0.0.1:62343 to
> > > > 127.0.0.1:19999
> > > > Deb2: Unicast : We create a client associated with the socket 64
> > > > Deb2: Unicast : first client
> > > > Deb2: Unicast : Actual SO_SNDBUF size : 2626560
> > > > Deb2: Unicast : Number of clients : 1
> > > > Deb2: Unicast : New message for socket 64
> > > > Deb2: Unicast : beginning of buffer GET /m
> > > > Deb2: Unicast : beginning of buffer 71 69 84 32 47 109
> > > > Deb2: Unicast : We received 94, buffer len 100 new buffer pos 94
> > > > Deb2: Unicast : End of HTTP request, we parse it
> > > > Deb0: Unicast : EIT Json
> > > > **\* Error in `/root/mumudvb': free(): invalid next size (fast):
> > > > 0x000000000063e540 ***
> > > > 
> > > > Program received signal SIGABRT, Aborted.
> > > > 0x00007ffff6adfcc9 in raise () from /lib/x86_64-linux-gnu/libc.so.6
> > > > (gdb) bt
> > > > #0 0x00007ffff6adfcc9 in raise () from
> > > > /lib/x86_64-linux-gnu/libc.so.6
> > > > #1 0x00007ffff6ae30d8 in abort () from
> > > > /lib/x86_64-linux-gnu/libc.so.6
> > > > #2 0x00007ffff6b1c394 in ?? () from /lib/x86_64-linux-gnu/libc.so.6
> > > > #3 0x00007ffff6b2866e in ?? () from /lib/x86_64-linux-gnu/libc.so.6
> > > > #4 0x00000000004088ec in convert_en300468_string (
> > > > string=string@entry=0x7ffffff2b2c0 "uf und davon - Mein
> > > > Auslandstagebuch", max_len=max_len@entry=4096,
> > > > debug=debug@entry=0) at log.c:1170
> > > > #5 0x0000000000418e4c in eit_show_short_evt_descr (
> > > > buf=buf@entry=0x77f759 "M+deu&\005Auf und davon - Mein
> > > > Auslandstagebuch",
> > > > reply=reply@entry=0xe591e0)
> > > > at unicast_EIT.c:476
> > > > #6 0x0000000000419579 in eit_display_descriptor (
> > > > buf=buf@entry=0x77f759 "M+deu&\005Auf und davon - Mein
> > > > Auslandstagebuch",
> > > > descriptors_loop_len=407,
> > > > reply=reply@entry=0xe591e0) at unicast_EIT.c:282
> > > > #7 0x000000000041997b in eit_display_contents
> > > > (full_eit=full_eit@entry=0x77f2e0,
> > > > 
> > > > reply=reply@entry=0xe591e0) at unicast_EIT.c:239
> > > > #8 0x0000000000419a09 in unicast_send_EIT_section
> > > > (eit_section=0x77f2e0,
> > > > num=num@entry=80,
> > > > reply=reply@entry=0xe591e0) at unicast_EIT.c:37
> > > > #9 0x000000000041cab3 in unicast_send_EIT
> > > > (eit_packets=eit_packets@entry=0x6844c0,
> > > > Socket=64)
> > > > at unicast_monit.c:259
> > > > #10 0x00000000004178eb in unicast_handle_message
> > > > (unicast_vars=unicast_vars@entry=0x7ffffff36150,
> > > > client=0xe59330, channels=channels@entry=0x7ffffff38548,
> > > > number_of_channels=number_of_channels@entry=12,
> > > > strengthparams=strengthparams@entry=0x7ffffff2d9b0,
> > > > auto_p=auto_p@entry=0x7ffffff36de0,
> > > > cam_p=cam_p@entry=0x7ffffff364d0,
> > > > scam_vars=scam_vars@entry=0x7ffffff36080,
> > > > eit_packets=eit_packets@entry=0x6844c0)
> > > > at unicast_http.c:818
> > > > #11 0x0000000000417c5f in unicast_handle_fd_event
> > > > (unicast_vars=unicast_vars@entry=0x7ffffff36150,
> > > > channels=channels@entry=0x7ffffff38548, number_of_channels=12,
> > > > strengthparams=strengthparams@entry=0x7ffffff2d9b0,
> > > > auto_p=auto_p@entry=0x7ffffff36de0,
> > > > 
> > > > cam_p=cam_p@entry=0x7ffffff364d0,
> > > > scam_vars=scam_vars@entry=0x7ffffff36080,
> > > > eit_packets=0x6844c0)
> > > > at unicast_http.c:391
> > > > #12 0x0000000000406005 in main (argc=<optimized out>,
> > > > argv=<optimized
> > > > out>) at mumudvb.c:1289
> > > > (gdb)
> > > > 
> > > > —
> > > > You are receiving this because you modified the open/close state.
> > > > Reply to this email directly or view it on GitHub
> > > > 
> > > > https://github.com/braice/MuMuDVB/issues/100#issuecomment-216003337
> > > 
> > > —
> > > You are receiving this because you authored the thread.
> > > Reply to this email directly or view it on GitHub
> > > https://github.com/braice/MuMuDVB/issues/100#issuecomment-216081348
> > 
> > —
> > You are receiving this because you modified the open/close state.
> > Reply to this email directly or view it on GitHub
> > https://github.com/braice/MuMuDVB/issues/100#issuecomment-216400990
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/braice/MuMuDVB/issues/100#issuecomment-216410893

Deb2:  TS:  Continuing a packet PID 18 cc 1 len 1471 expected 3618
Deb2:  TS:  General information PID 18 adaptation_field_control 1 payload_unit_start_indicator 0 continuity_counter 2
Deb2:  TS:  Continuing packet, data left 184
Deb2:  TS:  Continuing a packet PID 18 cc 2 len 1655 expected 3618
Deb2:  TS:  General information PID 18 adaptation_field_control 1 payload_unit_start_indicator 0 continuity_counter 3
Deb2:  TS:  Continuing packet, data left 184
Deb2:  TS:  Continuing a packet PID 18 cc 3 len 1839 expected 3618
Deb2:  TS:  General information PID 18 adaptation_field_control 1 payload_unit_start_indicator 0 continuity_counter 4
Deb2:  TS:  Continuing packet, data left 184
Deb2:  TS:  Continuing a packet PID 18 cc 4 len 2023 expected 3618
Deb2:  TS:  General information PID 18 adaptation_field_control 1 payload_unit_start_indicator 0 continuity_counter 5
Deb2:  TS:  Continuing packet, data left 184
Deb2:  TS:  Continuing a packet PID 18 cc 5 len 2207 expected 3618
Deb2:  TS:  General information PID 18 adaptation_field_control 1 payload_unit_start_indicator 0 continuity_counter 6
Deb2:  TS:  Continuing packet, data left 184
Deb2:  TS:  Continuing a packet PID 18 cc 6 len 2391 expected 3618
Deb2:  TS:  General information PID 18 adaptation_field_control 1 payload_unit_start_indicator 0 continuity_counter 7
Deb2:  TS:  Continuing packet, data left 184
Deb2:  TS:  Continuing a packet PID 18 cc 7 len 2575 expected 3618
Deb2:  TS:  General information PID 18 adaptation_field_control 1 payload_unit_start_indicator 0 continuity_counter 8
Deb2:  TS:  Continuing packet, data left 184
Deb2:  TS:  Continuing a packet PID 18 cc 8 len 2759 expected 3618
Deb2:  TS:  General information PID 18 adaptation_field_control 1 payload_unit_start_indicator 0 continuity_counter 9
Deb2:  TS:  Continuing packet, data left 184
Deb2:  TS:  Continuing a packet PID 18 cc 9 len 2943 expected 3618
Deb2:  TS:  General information PID 18 adaptation_field_control 1 payload_unit_start_indicator 0 continuity_counter 10
Deb2:  TS:  Continuing packet, data left 184
Deb2:  TS:  Continuing a packet PID 18 cc 10 len 3127 expected 3618
Deb2:  Unicast :  New client
Deb2:  Unicast :  New connection from 127.0.0.1:15308 to 127.0.0.1:19999
Deb2:  Unicast :  We create a client associated with the socket 64
Deb2:  Unicast :  first client
Deb2:  Unicast :  Actual SO_SNDBUF size : 2626560
Deb2:  Unicast :  Number of clients : 1
Deb2:  TS:  General information PID 18 adaptation_field_control 1 payload_unit_start_indicator 0 continuity_counter 11
Deb2:  TS:  Continuing packet, data left 184
Deb2:  TS:  Continuing a packet PID 18 cc 11 len 3311 expected 3618
Deb2:  Unicast :  New message for socket 64
Deb2:  Unicast :  beginning of buffer GET /m
Deb2:  Unicast :  beginning of buffer 71 69 84 32 47 109
Deb2:  Unicast :  We received 94, buffer len 100 new buffer pos 94
Deb2:  Unicast :  End of HTTP request, we parse it
Deb0:  Unicast :  EIT Json
**\* Error in `/root/mumudvb': malloc(): memory corruption: 0x0000000000ed8be0 ***

Program received signal SIGABRT, Aborted.
0x00007ffff6adfcc9 in raise () from /lib/x86_64-linux-gnu/libc.so.6
(gdb) bt
#0  0x00007ffff6adfcc9 in raise () from /lib/x86_64-linux-gnu/libc.so.6
#1  0x00007ffff6ae30d8 in abort () from /lib/x86_64-linux-gnu/libc.so.6
#2  0x00007ffff6b1c394 in ?? () from /lib/x86_64-linux-gnu/libc.so.6
#3  0x00007ffff6b29f36 in ?? () from /lib/x86_64-linux-gnu/libc.so.6
#4  0x00007ffff6b2b7b0 in malloc () from /lib/x86_64-linux-gnu/libc.so.6
#5  0x00007ffff6ad56d9 in ?? () from /lib/x86_64-linux-gnu/libc.so.6
#6  0x00007ffff6acd3da in ?? () from /lib/x86_64-linux-gnu/libc.so.6
#7  0x00007ffff6acbd5f in ?? () from /lib/x86_64-linux-gnu/libc.so.6
#8  0x00007ffff6acb8e2 in iconv_open () from /lib/x86_64-linux-gnu/libc.so.6
#9  0x000000000040889f in convert_en300468_string (string=string@entry=0x7ffffff2b2c0 "",
    max_len=max_len@entry=4096, debug=debug@entry=0) at log.c:1153
#10 0x0000000000418e4c in eit_show_short_evt_descr (
    buf=buf@entry=0xddd78a "M)deu$\005hundkatzemaus / Das Haustiermagazin", reply=reply@entry=0xa9e0f0)
    at unicast_EIT.c:476
#11 0x0000000000419579 in eit_display_descriptor (
    buf=buf@entry=0xddd78a "M)deu$\005hundkatzemaus / Das Haustiermagazin", descriptors_loop_len=577,
    reply=reply@entry=0xa9e0f0) at unicast_EIT.c:282
#12 0x000000000041997b in eit_display_contents (full_eit=full_eit@entry=0xddd770,
    reply=reply@entry=0xa9e0f0) at unicast_EIT.c:239
#13 0x0000000000419a09 in unicast_send_EIT_section (eit_section=0xddd770, num=num@entry=40,
    reply=reply@entry=0xa9e0f0) at unicast_EIT.c:37
#14 0x000000000041cab3 in unicast_send_EIT (eit_packets=eit_packets@entry=0x6844c0, Socket=64)
    at unicast_monit.c:259
#15 0x00000000004178eb in unicast_handle_message (unicast_vars=unicast_vars@entry=0x7ffffff36150,
    client=0xa9dfe0, channels=channels@entry=0x7ffffff38548,
    number_of_channels=number_of_channels@entry=12, strengthparams=strengthparams@entry=0x7ffffff2d9b0,
    auto_p=auto_p@entry=0x7ffffff36de0, cam_p=cam_p@entry=0x7ffffff364d0,
    scam_vars=scam_vars@entry=0x7ffffff36080, eit_packets=eit_packets@entry=0x6844c0)
    at unicast_http.c:818
#16 0x0000000000417c5f in unicast_handle_fd_event (unicast_vars=unicast_vars@entry=0x7ffffff36150,
    channels=channels@entry=0x7ffffff38548, number_of_channels=12,
    strengthparams=strengthparams@entry=0x7ffffff2d9b0, auto_p=auto_p@entry=0x7ffffff36de0,
    cam_p=cam_p@entry=0x7ffffff364d0, scam_vars=scam_vars@entry=0x7ffffff36080, eit_packets=0x6844c0)
    at unicast_http.c:391
#17 0x0000000000406005 in main (argc=<optimized out>, argv=<optimized out>) at mumudvb.c:1289
(gdb)

**Comment by braice** [comment]
Hi

Can you try with the attached log.c

If it doesn't work do you think it is possible to give me access to your machine ?
[log.c.zip](https://github.com/braice/MuMuDVB/files/251352/log.c.zip)

**Comment by SourceDoctor** [comment]
HI,

with this log.c i can fetch EIT Json.

But now there are Characters in Json which makes it incompatible

In _extended Descriptor Text Fields_ there are now single "\" printed
before a Newline Character.

{
        "tag" :78,
        "len" : 218,
        "descr" : "Extended event descriptor",
        "ext_evt":{
        "descr_number" : 2 ,
        "descr_total_number" : 2,
        "language" : "DEU",
        "items":[],
        "text": "enn Etikette ist auch heute immer noch das A und O ...
Wer soll da noch mithalten? Ständig ein neuer Szene-Drink!\
_\_
_\_
Darsteller:_\_
Shirin Soraya_\_
Nina Vorbrodt_\_
Emily Wood_\_
Hanno Friedrich_\_
Thomas M. Held_\_
Mirco Reseg"
}

In *Component descriptor in Text Fields *the last Character now often
seems to be a control Character, which also breaks JSON:

{
        "tag" :80,
        "len" : 15,
        "descr" : "Component descriptor",
        "component":{
        "len" : 13,
        "stream_content" :2,
        "component_type" :3,
        "component_tag" :3,
        "language" : "DEU",
        "text" : "Stereo_^L_"
}

As Control Charakter only Newline for keeping the Textstructure, is
interesting i think.

Am 06.05.2016 um 01:21 schrieb Brice Dubost:

**Comment by braice** [comment]
Hello

Thank you for your report, at least it goes forward

Following your e-mail I noticed a stupid mistake in escaping control
characters

I corrected it, can you give it a try ?

Thank you

2016-05-08 7:50 GMT-04:00 SourceDoctor notifications@github.com:

**Comment by SourceDoctor** [comment]
Hi Brice,

i tested it and it runs well, including with newlines in it, thanks for
support and this software itself btw :-) .

But i recognized an other Failure which seems to come up now.

On many Programs in short-evt and extended-evt now the first Charakter
gets lost,
also there is often an unwished Charakter in component descriptor text.

Here an example.

"EIT_events":[
{
                "event_id" :43987,
                "start_time day " : "2016-05-22 (yy-mm-dd)",
                "start_time" : "02:05:00",
                "duration" : "01:55:00",
                "running_status" :0,
                "running_status_descr" : "undefined",
                "free_ca_mode" :0,
                "descriptors_loop_length" : 45,
"EIT_descriptors":[

{
                "tag" :77,
                "len" : 22,
                "descr" : "Short event descriptor",
                "short_evt":{
                "language" : "deu",
                "name" : "ARD - Hitnacht",
                "text" : "RD - Hitnacht"
_--(SHOULD BE)>>>>                "text" : "ARD - Hitnacht"_
}
},
{
                "tag" :80,
                "len" : 14,
                "descr" : "Component descriptor",
                "component":{
                "len" : 12,
                "stream_content" :2,
                "component_type" :3,
                "component_tag" :2,
                "language" : "deu",
                "text" : "stereoT"
_--(SHOULD BE)>>>>                "text" : "stereo"_
}
},
{
                "tag" :84,
                "len" : 4,
                "descr" : "Content descriptor (type of program cf tabl
28 EN 300 468 V1.13.1)",
                "content":{
                "content_nibble_lvl_1" :11,
                "content_nibble_lvl_2" :15,
                "user byte" :203
}
},
{
                "tag" :105,
                "len" : 5,
                "descr" : "PDC descriptor"
}]

Am 15.05.2016 um 03:18 schrieb Brice Dubost:

**Comment by braice** [comment]
Hi

Ok it seems that I have still some issues counting characters properly,
I just have made a branch called debug with messages to test this

Can you run it in very verbose from this branch, provide me the log as well
as you just did (the failing JSON)

Thank you

2016-05-18 18:24 GMT-04:00 SourceDoctor notifications@github.com:

**Comment by SourceDoctor** [comment]
Hi Brice,

something found in the Logs i mailed you to mumudvb@braice.net?

with regards

   Thomas

Am 23.05.2016 um 02:18 schrieb Brice Dubost:

**Comment by m0yellow** [comment]
I love to follow you guys, but can you please cut the crap (literally) from your e-Mails? much appreciated!
